### PR TITLE
Sysinfo: cleanups and preps for nrunner

### PR DESCRIPTION
**Pull request by clebergnu**
While reviewing #4806, I had the impression that less code needed to be duplicated between the nrunner-based and the current sysinfo implementation.  This is an exercise at that, which also results in cleanups and overall improvements (shamelessly copied from @richtja's work).

### CODE REVIEW

File: avocado/core/sysinfo.py [hunk 1]
```diff
@@ -44,6 +44,22 @@ def __init__(self, log_path):
     def run(self, logdir):
         pass
 
+    def __eq__(self, other):
+        if hash(self) == hash(other):
+            return True
+        elif isinstance(other, Collectible):
+            return False
+        return NotImplemented
```
Comment (beraldoleal) [inline avocado/core/sysinfo.py:52]: NotImplemented or false?

**Review by beraldoleal — COMMENTED** [review]
Hi @clebergnu , this LGTM. just one small comment.

File: avocado/core/sysinfo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (clebergnu) [reply avocado/core/sysinfo.py:52]: The reason is that, by returning `NotImplemented`, we delegate the type checking to other (upper, side?) inheritance.  See https://docs.python.org/3/library/constants.html#NotImplemented

[the remark this answers is not in the document]
Comment (beraldoleal) [reply avocado/core/sysinfo.py:52]: That is what I was missing. thanks

**Review by beraldoleal — APPROVED** [review]
LGTM.

**Pull request merged**